packages/flutter_markdown/lib/src/builder.dart

```diff
@@ -85,6 +85,84 @@ abstract class MarkdownBuilderDelegate {
   TextSpan formatText(MarkdownStyleSheet styleSheet, String code);
 }
 
+/// An anchor in markdown is basically a marker inside the markdown text
+/// with an id.
+/// We can use this to scroll to a specific place inside the text.
+///
+/// When using [md.ExtensionSet.gitHubWeb] (more specifically
+/// [md.HeaderWithIdSyntax] and [md.SetextHeaderWithIdSyntax]) anchors are
+/// automatically created for headings.
+///
+/// We can then use the [AnchorController] to scroll to a specific header and
+/// see which headers are currently visible.
+///
+/// Usually one can also create anchors in markdown via the html tag `a` but as
+/// we can't render HTML inside flutter currently (as far as I know) only
+/// headers should have an anchor.
+class AnchorData {
+  AnchorData(this.id, this.text);
+
+  /// The id of the anchor e.g. 'table-of-contents'.
+  final String id;
+
+  /// The name of the anchor (so the text of the heading if the anchor is for a
+  /// heading).
+  /// E.g. 'Table of contents'.
+  final String text;
```

Comment:
Since the `Element.textContent` from `markdown` is non-nullable I also return a non-nullable String here.  
As in flutter they only way (I think!) to create an anchor right now is to use the generated Ids for headings via e.g. `ExtensionSet.githubWeb` there should always be a `text`.     
I don't like this really though as the concept of an anchor doesn't require headings or a text. Should this be nullable then even though (right now) it never can be null?

Reply:
Would an empty string be acceptable for an unnamed anchor? 

If not, we should definitely consider making this nullable to avoid the need for a breaking change in the future.

Reply:
Well I mean I'm not sure.   

The `markdown` package implements `Element.textContent` like this:  
```dart
  @override
  String get textContent {
    return (children ?? []).map((child) => child.textContent).join('');
  }
```
So instead of making it nullable it returns an empty String.   

So one question for me is if we should just follow that behavior.   
The other question for me is if there might be any way in the future for the `markdown` or `flutter_markdown` plugin to have an anchor without a text (so e. g. interpreting `a` html tags as an anchor inside flutter) and thus a reason why we should make it nullable anyways. Right now there should always be the heading text (I think).